docs/developing/app-review-guide.md

```diff
@@ -0,0 +1,96 @@
+
+The reviewer may look at the changelog and commit history to narrow down
+what new features or changes should be tested, and may reach out to the
+developer with suggestions or feedback. However, in many cases, the app reviewer
+will simply approve the update after verifying its functionality and notify the
+developer that it has been approved.
+
+## Approval-blocking issues
+
+The Sandstorm App Market is a fairly permissive environment. While you may
+receive feedback about performance, functionality, or usability, we largely
+require only that your app works and that it does not lose data.
+
+We will reject an app which crashes immediately and does not function as
+advertised, for example. This can often occur when the package is missing files
+that were present in the dev environment and wasn't adequately tested prior to
+submission.
+
+The latter concern often occurs when a pure client app is submitted up that
+implements localStorage as its storage medium, which Sandstorm does not effectively
+support. This can often be tested by storing some information in a grain, allowing
+Sandstorm to shut down that grain, and then opening it again and seeing if things
+persisted that an ordinary user would expect to persist between sessions.
+
+Finally, the App Market may reject apps which present security or privacy issues
+that fall outside of what a Sandstorm user would expect when using the app. This
+can mean it requests overly-broad permissions when making Powerbox requests or
+uses client-side loading to load scripts which share unexpected information outside
+of the Sandstorm sandbox. For example, we may block approval on an app which attempts
+to load an analytics script from an outside server, or does not make it suitably
+clear to a user why it needs certain outside network access prior to requesting it.
+
+**Note:** Client-side loading is a known gap in Sandstorm's security model that we
+intend to close. Not only is misuse of it a security or privacy issue, its use may
+lead to apps that break when Sandstorm closes this gap.
+
+## Commercial or proprietary apps
+
+The Sandstorm App Market permits both commercial and proprietary apps, however
+official support for app purchase and licensing is not currently available. For an
+app that is merely closed source, but free, this may not pose an issue. For apps
+which require some form of licensing to function, we expect at minimum that the
+app description clearly and prominently specifies the licensing requirements and
+activation method. We would also encourage the app to have at least a minimal level of
+functionality such as a trial or limited version, such that users can experience the
+app before deciding on a purchase. As this is a territory the Sandstorm app review
+team hasn't spent a lot of time on, interested parties may wish to reach out to the
+Sandstorm community for feedback prior to committing significant work here.
```

Comment:
So I'm still unhappy with this section. I'm not as worried from a legal perspective if we just keep it to the blurb @kentonv outlined below. But I think it tries too hard to accommodate anticipated industry expectations around DRM that nobody's even asked us about yet. I don't think we should be making concessions before we've gotten any pushback at all. Also the wording here is a bit confusing in that you use the word "licensing" when what I think you mean is DRM.

Here's an alternate attempt at this section:

---

The Sandstorm App Market permits both commercial and proprietary apps. However,
they are not our primary focus. Developers of such apps should be aware that:
 -  The App "Market" is missing some features you might expect from platforms
built around proprietary software, such as a payment system of any kind, or DRM/copy
protection.
- Since most apps in the app market are currently Free & Open Source, users may be
surprised to find that yours isn't; you should display this fact clearly and prominently.
 - Sandstorm as a project focused on user autonomy. This creates an inherent conflict
with DRM or any other technical mechanism for enforcing licensing requirements,
and technical decisions made in building the platform will always favor user
autonomy over developer control. Furthermore, we do not consider DRM/license
enforcement to be a valid reason for requesting broader permissions than would
otherwise be required; see the section above regarding approval blocking issues.
If you feel you need to implement some kind of DRM for your app, talk to us first,
before investing time and effort into a solution that we may not find acceptable.

---

At some point I'd like to have a more well defined "user's bill of rights" or such that
defines what things we really aren't willing to budge on, but for the time being I'd rather
err on the side of strongly discouraging doing anything funny without talking to us.